src/cmakeProject.ts

```diff
@@ -2674,10 +2670,7 @@ export class CMakeProject {
         if (path.basename(expandedSourceDirectory).toLocaleLowerCase() !== "cmakelists.txt") {
             expandedSourceDirectory = path.join(expandedSourceDirectory, "CMakeLists.txt");
         }
-
-        this.hasCMakeListsFile = await fs.exists(expandedSourceDirectory);
-
-        return this.hasCMakeListsFile;
+        return fs.exists(expandedSourceDirectory);
```

Comment:
It might be good for us to come back and make this more efficient in the future (or do it now).  For example, whenever we set `this._sourceDir` for the CMakeProject we could check the file exists and set `hasCMakeListsFile` at that time so we don't have to do any extra work in this function.  This function should probably have been calling `util.normalizeAndVerifySourceDir` instead of doing this custom work for the setting to populate `expandedSourceDirectory`.